controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1855,6 +1860,27 @@ func (r *BareMetalHostReconciler) createHostFirmwareSettings(info *reconcileInfo
 	return nil
 }
 
+func (r *BareMetalHostReconciler) hostUpdatePolicySetOwnerReference(info *reconcileInfo) error {
+	// NOTE(janders) the goal here is to ensure that the controller reads the hup resource and adds OwnerReference to it
+	hup := &metal3api.HostUpdatePolicy{}
+	if err := r.Get(info.ctx, info.request.NamespacedName, hup); err != nil {
+		// Error reading the object
+		return errors.Wrap(err, "could not load hostUpdatePolicy resource")
```

Comment:
Please don't use the deprecated errors.Wrap, you need fmt.Errorf

Reply:
Done, thank you for the pointer.